hail/src/main/scala/is/hail/expr/ir/MatrixWriter.scala

```diff
@@ -972,8 +973,13 @@ case class MatrixBGENWriter(
     else
       path + ".bgen"
 
+    assert(compressionCodec == "zlib" || compressionCodec == "zstd")
     val writeHeader = exportType == ExportType.PARALLEL_HEADER_IN_SHARD
-    val partWriter = BGENPartitionWriter(tm, entriesFieldName, writeHeader)
+    val compressionInt = compressionCodec match {
```

Comment:
Is there any way to make this some kind of enum, or if that's too much, named constants?

Reply:
done.